datafusion/core/src/execution/context.rs

```diff
@@ -318,6 +319,19 @@ impl SessionContext {
         let plan = self.state().create_logical_plan(sql).await?;
 
         match plan {
+            LogicalPlan::Dml(DmlStatement {
+                table_name,
+                op: WriteOp::Insert,
+                input,
+                ..
+            }) => {
+                if self.table_exist(&table_name)? {
```

Comment:
This should return an error if the table does not exist.

Reply:
Certainly, I will rectify it.

Reply:
Addressed 👍🏻